### PR TITLE
Support TypeScript without ember-cli-typescript

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Repro'd and then patched here: https://github.com/NullVoxPopuli/ember-demo-typescript-without-ec-ts/commit/dd5baa622e2c564ec150a9f0b4f028ac24f8c9bf

With glint + `<template>`, ember-cli-typescript's type-checking worker is incorrect as tsserver can't, by itself, resolve .gts imports. 

Since glint projects fully manage type checking within glint (and gts transformation via ember-template-imports), ember-cli-typescript should be removed as a requirement for projects using typescript so that we can reduce terminal noise in the build.

Later, we may bundle glint in some webpack (or preferrably, [unplugin](https://github.com/unjs/unplugin)) plugin such that we properly type check ember projects with glint / template-imports (maybe similar to [`fork-ts-checker--webpack-plugin`](https://www.npmjs.com/package/fork-ts-checker-webpack-plugin) )

anywho, this PR is mostly tests:
 - additional app template, based off of the ember-cli-typescript blueprint
   but also
   - some simplifications of the tsconfig.json
   - convert app.js, router.js, and test-helper.js to ts to have a minimal set of "ts stuff" for every test/scenario
 
Resolves: https://github.com/embroider-build/embroider/issues/889

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
I probably broke something :sweat_smile: 

```
Build Error (PackagerRunner) in assets/test.js

Module build failed (from ../../../../../..<repo>/tests/scenarios/output/node_modules/@embroider/webpack/node_modules/thread-loader/dist/cjs.js):
Thread Loader (Worker 1)
[BABEL] $TMPDIR/embroider/d9b711/tests/scenarios/output/assets/test.js/test.js: module.call is not a function (While processing: "<repo>/packages/core/src/portable-babel-launcher.js")

    at babelLauncher (<repo>/packages/core/src/portable-babel-launcher.js:18:25)}
```

File: packages/compat/src/compat-app.ts [hunk 1]
```diff
@@ -233,23 +233,11 @@ class CompatAppAdapter implements AppAdapter<TreeNames> {
...
-    // resolvable and there's no other "loader" or anything to apply.
-    if (this.activeAddonChildren().find(pkg => pkg.name === 'ember-cli-typescript')) {
-      extensions.unshift('.ts');
-    }
-
-    return extensions;
+    // For TS, we defer to ember-cli-babel, and the setting for
+    // "enableTypescriptTransform" can be set with and without
+    // ember-cli-typescript
+    return ['.wasm', '.mjs', '.js', '.json', '.hbs', '.ts'];
```
Comment (bwbuchanan) [inline packages/compat/src/compat-app.ts:240]: This introduces a bug that causes TypeScript component files to be ignored when `staticComponents: true`

should be: `return ['.ts', '.wasm', '.mjs', '.js', '.json', '.hbs'];`

Comment (NullVoxPopuli) [reply packages/compat/src/compat-app.ts:240]: how does that happen / how does moving ts to the front resolve it?

Comment (bwbuchanan) [reply packages/compat/src/compat-app.ts:240]: I'm not at all familiar with the internals of embroider and had to diagnose this by trial-and-error.

The previous version of the code unshifts '`.ts'` onto the array (placing it at the front). This change moves it to the end of the array.

When I inspected $TMPDIR/embroider/.../components, I found three files for each of my TypeScript components: .hbs, .js, and .ts.

.hbs and .ts were as expected. .js contains a templateOnlyComponent stub.

I can only deduce that in the process of assembling the app, embroider looks for the component implementation file in the order of the extensions in this array, so moving .ts to the end of the array means that it finds the empty stub implementation and ignores the actual implementation in the .ts file.

I have no idea why the stub .js implementations are being generated if a .ts backing class definition exists. Perhaps this is a different bug, or perhaps intended behavior?

The result is a broken component with no backing class.

Doesn't happen if staticComponents = false in the embroider config.

**Pull request merged**